### DISCUSSION TITLE
Filtering out onmicrosoft accounts in the react-directory webpart

### DISCUSSION THREAD

**Discussion by johnsoep** [pnp/sp-dev-fx-webparts#5161] [Q&A]
I'm working on seeing if we can utilize the react-directory webpart for my organization. I work in at a State University and we have to migrate our campus directory solution to a new platform within a year. We're also working on implementing a new Intranet utilizing SharePoint so I am really pushing to get as much into our SharePoint environment as we can and I think that the react-directory webpart meets most of our needs.

The one adjustment that I need to make is to get the webpart to ignore the onmicrosoft accounts. We have our environment setup to work off our domain, but when I do a search in the react-directory webpart it is showing Microsoft Bookings and M365 local accounts. I would like to adjust the query to only return our domain.edu addresses but I'm not 100% sure where to put that. My TypeScript is pretty basic and I don't have a lot of programming experience in the SharePoint Web Part world. Any feedback or tips would be greatly appreciated!

**Comment by netmatrixtech** [comment]
Please see that I've created my own enhancements to react-directory to facilitate filtering of results based on Entra ID Group Object ID. It'll be up to the contributors for this webpart to decide if they wish to add it into their project.
It should cater to most of your requests if you use an Entra Dynamic Group to apply the accounts you want shown in the results of the webpart.

[https://github.com/pnp/sp-dev-fx-webparts/discussions/6463](https://github.com/pnp/sp-dev-fx-webparts/discussions/url)
cheers
Tim
